### PR TITLE
[NAE-1591] Unit tests failing on "An error was thrown in afterAll"

### PR DESCRIPTION
**Pull request by Kovy95**
# Description

- add to all tests TestBed.resetTestingModule()
- add done to tests with subscribe
- add done to tests with mocks

Fixes [NAE-1591]

## Dependencies

None.

### Third party dependencies

No new dependencies were introduced

### Blocking Pull requests

There are no dependencies on other PR

## How Has Been This Tested?

3 x 16 npm run ncc:full-test without error

### Test Configuration

| Name                | Tested on |
|---------------------| --------- |
| OS                  |    Linux Mint 20.3       |
| Runtime             |  Node.js 14.19.0         |
| Dependency Manager  |  NPM 6.14.16         |
| Framework version   |           |
| Run parameters      |           |
| Other configuration |           |

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes have been checked, personally or remotely, with @...
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have resolved all conflicts with the target branch of the PR
- [x] I have updated and synced my code with the target branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes:
    - [x] Lint test
    - [x] Unit tests
    - [x] Integration tests
- [x] I have checked my contribution with code analysis tools:
    - [x] [SonarCloud](https://sonarcloud.io/project/overview?id=netgrif_application-engine)
    - [x] [Snyk](https://app.snyk.io/org/netgrif)
- [x] I have made corresponding changes to the documentation:
    - [x] Developer documentation
    - [x] User Guides
    - [x] Migration Guides

[NAE-1591]: https://netgrif.atlassian.net/browse/NAE-1591?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: projects/netgrif-components-core/src/lib/search/search-predicate-component/abstract-search-predicate.component.spec.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import {LoggerService} from '../../logger/services/logger.service';
 import {CategoryFactory} from '../category-factory/category-factory';
 import {ConfigurationService} from '../../configuration/configuration.service';
 import {TestConfigurationService} from '../../utility/tests/test-config';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
Comment (minop) [inline projects/netgrif-components-core/src/lib/search/search-predicate-component/abstract-search-predicate.component.spec.ts:10]: why was this import added?

File: projects/netgrif-components-core/src/lib/search/search-service/search.service.spec.ts [hunk 2]
```diff
@@ -1,11 +1,11 @@
-import {TestBed} from '@angular/core/testing';
+import {inject, TestBed} from '@angular/core/testing';
 import {SearchService} from './search.service';
 import {NoopAnimationsModule} from '@angular/platform-browser/animations';
 import {CategoryFactory} from '../category-factory/category-factory';
 import {CaseTitle} from '../models/category/case/case-title';
 import {ConfigurationService} from '../../configuration/configuration.service';
 import {TestConfigurationService} from '../../utility/tests/test-config';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
Comment (minop) [inline projects/netgrif-components-core/src/lib/search/search-service/search.service.spec.ts:8]: why was this import added?

File: projects/netgrif-components-core/src/lib/side-menu/content-components/import-net/abstract-import-net.component.spec.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import {waitForAsync, ComponentFixture, TestBed} from '@angular/core/testing';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
Comment (minop) [inline projects/netgrif-components-core/src/lib/side-menu/content-components/import-net/abstract-import-net.component.spec.ts:2]: why was this import added?

File: projects/netgrif-components-core/src/lib/side-menu/side-menu-container/abstract-side-menu-container.component.spec.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
-import {waitForAsync, ComponentFixture, TestBed} from '@angular/core/testing';
+import {waitForAsync, ComponentFixture, TestBed, inject} from '@angular/core/testing';
 import {CommonModule} from '@angular/common';
 import {NoopAnimationsModule} from '@angular/platform-browser/animations';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
Comment (minop) [inline projects/netgrif-components-core/src/lib/side-menu/side-menu-container/abstract-side-menu-container.component.spec.ts:4]: why was this import added?

File: projects/netgrif-components-core/src/lib/task/services/delegate-task.service.spec.ts [hunk 5]
```diff
@@ -3,7 +3,7 @@ import {DelegateTaskService} from './delegate-task.service';
 import {NoopAnimationsModule} from '@angular/platform-browser/animations';
 import {ConfigurationService} from '../../configuration/configuration.service';
 import {TestConfigurationService} from '../../utility/tests/test-config';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
Comment (minop) [inline projects/netgrif-components-core/src/lib/task/services/delegate-task.service.spec.ts:6]: why was this import added?

File: projects/netgrif-components-core/src/lib/task/services/task-data.service.spec.ts [hunk 6]
```diff
@@ -1,8 +1,8 @@
-import {TestBed} from '@angular/core/testing';
+import {inject, TestBed} from '@angular/core/testing';
 import {TaskDataService} from './task-data.service';
 import {TaskRequestStateService} from './task-request-state.service';
 import {TranslateLibModule} from '../../translate/translate-lib.module';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
Comment (minop) [inline projects/netgrif-components-core/src/lib/task/services/task-data.service.spec.ts:5]: why was this import added?

File: projects/netgrif-components-core/src/lib/authentication/services/anonymous-authentication-interceptor.spec.ts [hunk 7]
```diff
@@ -33,22 +31,28 @@ describe('AnonymousAuthenticationInterceptor', () => {
...
-                http.get('/api').subscribe(response => {
-                    expect(response).toBeTruthy();
-                });
-                const request = mock.expectOne(req => (req.headers.has('X-Jwt-Token')));
+                    service.setToken('jwt-token');
+                    http.get('/api').subscribe(response => {
+                        expect(response).toBeTruthy();
+                        done();
+                    });
+                    const request = mock.expectOne(req => (req.headers.has('X-Jwt-Token')));
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/authentication/services/anonymous-authentication-interceptor.spec.ts:43]: yes, because subscription is called after request.flush

File: projects/netgrif-components-core/src/lib/header/case-header/case-header.service.spec.ts [hunk 8]
```diff
@@ -139,8 +140,8 @@ describe('CaseHeaderService', () => {
                 expect(res.changeType).toEqual(HeaderChangeType.MODE_CHANGED);
                 expect((res.description as ModeChangeDescription).previousMode).toEqual(HeaderMode.EDIT);
                 expect((res.description as ModeChangeDescription).currentMode).toEqual(HeaderMode.SORT);
-                done();
             }
+            done();
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/header/case-header/case-header.service.spec.ts:144]: done

File: projects/netgrif-components-core/src/lib/process/process.service.spec.ts [hunk 9]
```diff
@@ -41,40 +41,54 @@ describe('ProcessService', () => {
         expect(service).toBeTruthy();
     });
 
-    it('should call one net', () => {
+    it('should call one net', (done) => {
         const getOneSpy = spyOn(TestBed.inject(PetriNetResourceService), 'getOne').and.callThrough();
-        service.getNet('correct').subscribe();
+        service.getNet('correct').subscribe(() => {
+            done();
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/process/process.service.spec.ts:47]: done

File: projects/netgrif-components-core/src/lib/process/process.service.spec.ts [hunk 10]
```diff
@@ -132,26 +146,30 @@ describe('ProcessService', () => {
...
+        });
         expect(logInfoSpy).toHaveBeenCalled();
     });
 
-    it('get petri net reference errors - roles fail to load', () => {
-        service.getNetReference('errorOnRoles').subscribe();
-        expect(logSpy).toHaveBeenCalled();
-    });
+    // it('get petri net reference errors - roles fail to load', (done) => {
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/process/process.service.spec.ts:156]: it will be uncommented with Angular 13 version, because of [Obecjt object ] thrown

File: projects/netgrif-components-core/src/lib/resources/engine-endpoint/case-resource.service.spec.ts [hunk 11]
```diff
@@ -23,136 +23,164 @@ describe('CaseResourceService', () => {
...
-        })
+    it('should count', (done) => {
+        inject([HttpTestingController],
+                (httpMock: HttpTestingController) => {
+                    service.count(new SimpleFilter('', FilterType.CASE, {})).subscribe(res => {
+                        expect(res.count).toEqual(0);
+                        done();
+                    });
+
+                    const reqLog = httpMock.expectOne('http://localhost:8080/api/workflow/case/count');
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/resources/engine-endpoint/case-resource.service.spec.ts:34]: it was personally consulted, done

File: projects/netgrif-components-core/src/lib/resources/engine-endpoint/dashboard-resource.service.spec.ts [hunk 12]
```diff
@@ -22,20 +22,24 @@ describe('DashboardResourceService', () => {
...
+        inject([HttpTestingController],
+                (httpMock: HttpTestingController) => {
+                    service.getDashboardData('case', {aggs: {result: {terms: {field: 'dataSet.text.value.keyword'}}}}).subscribe(res => {
+                        expect(res.aggregations).toEqual({});
+                        done();
+                    });
 
-            const reqLog = httpMock.expectOne('http://localhost:8080/api/dashboard/search?type=case');
-            expect(reqLog.request.method).toEqual('POST');
+                    const reqLog = httpMock.expectOne('http://localhost:8080/api/dashboard/search?type=case');
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/resources/engine-endpoint/dashboard-resource.service.spec.ts:33]: it was personally consulted, done

File: projects/netgrif-components-core/src/lib/resources/engine-endpoint/petri-net-resource.service.spec.ts [hunk 13]
```diff
@@ -20,124 +20,152 @@ describe('PetriNetResourceService', () => {
...
-        })
+    it('should getAll', (done) => {
+        inject([HttpTestingController],
+                (httpMock: HttpTestingController) => {
+                    service.getAll().subscribe(res => {
+                        expect(res.length).toEqual(0);
+                        done();
+                    });
+
+                    const reqLog = httpMock.expectOne('http://localhost:8080/api/petrinet');
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/resources/engine-endpoint/petri-net-resource.service.spec.ts:31]: it was personally consulted, done

File: projects/netgrif-components-core/src/lib/resources/engine-endpoint/task-resource.service.spec.ts [hunk 14]
```diff
@@ -22,189 +22,241 @@ describe('TaskResourceService', () => {
...
-        })
+    it('should count', (done) => {
+            inject([HttpTestingController],
+                (httpMock: HttpTestingController) => {
+                    service.count(new SimpleFilter('', FilterType.TASK, {})).subscribe(res => {
+                        expect(res.count).toEqual(0);
+                        done();
+                    });
+
+                    const reqLog = httpMock.expectOne('http://localhost:8080/api/task/count');
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/resources/engine-endpoint/task-resource.service.spec.ts:33]: it was personally consulted, done

File: projects/netgrif-components-core/src/lib/translate/translate-interceptor.spec.ts [hunk 15]
```diff
@@ -39,21 +39,22 @@ describe('TranslateInterceptor', () => {
...
 
-                http.get('/api').subscribe(response => {
-                    expect(response).toBeTruthy();
-                });
-                const request = mock.expectOne(req => req.headers.has('Accept-Language'));
+                    http.get('/api').subscribe(response => {
+                        expect(response).toBeTruthy();
+                        done();
+                    });
+                    const request = mock.expectOne(req => req.headers.has('Accept-Language'));
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/translate/translate-interceptor.spec.ts:50]: it was personally consulted, done

File: projects/netgrif-components-core/src/lib/view/case-view/service/case-view-service.spec.ts [hunk 16]
```diff
@@ -1,7 +1,7 @@
-import {fakeAsync, TestBed, tick} from '@angular/core/testing';
+import {fakeAsync, inject, TestBed, tick} from '@angular/core/testing';
 import {CaseViewService} from './case-view-service';
 import {ConfigurationService} from '../../../configuration/configuration.service';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/view/case-view/service/case-view-service.spec.ts:4]: done

File: projects/netgrif-components-core/src/lib/view/task-view/service/task-view.service.spec.ts [hunk 17]
```diff
@@ -1,6 +1,6 @@
-import {fakeAsync, TestBed, tick} from '@angular/core/testing';
+import {fakeAsync, inject, TestBed, tick} from '@angular/core/testing';
 import {ConfigurationService} from '../../../configuration/configuration.service';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/view/task-view/service/task-view.service.spec.ts:3]: done

File: projects/netgrif-components-core/src/lib/view/tree-case-view/tree-component/case-tree.service.spec.ts [hunk 18]
```diff
@@ -1,7 +1,7 @@
-import {TestBed} from '@angular/core/testing';
+import {inject, TestBed} from '@angular/core/testing';
 import {CaseTreeService} from './case-tree.service';
 import {TranslateLibModule} from '../../../translate/translate-lib.module';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/view/tree-case-view/tree-component/case-tree.service.spec.ts:4]: done

File: projects/netgrif-components-core/src/lib/authentication/services/authentication-interceptor.spec.ts [hunk 19]
```diff
@@ -34,22 +34,27 @@ describe('AuthenticationInterceptor', () => {
...
-                http.get('/api').subscribe(response => {
-                    expect(response).toBeTruthy();
-                });
-                const request = mock.expectOne(req => (req.headers.has('X-Auth-Token')));
+                    service.sessionToken = 'session-token';
+                    http.get('/api').subscribe(response => {
+                        expect(response).toBeTruthy();
+                        done();
+                    });
+                    const request = mock.expectOne(req => (req.headers.has('X-Auth-Token')));
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/authentication/services/authentication-interceptor.spec.ts:46]: yes, because subscription is called after request.flush

File: projects/netgrif-components-core/src/lib/authentication/services/methods/basic-authentication/basic-authentication.service.spec.ts [hunk 20]
```diff
@@ -36,36 +36,43 @@ describe('BasicAuthenticationService', () => {
...
-                expect(res.id).toEqual('id');
-            });
+                service.login({username: 'name', password: 'pass'}).subscribe(res => {
+                    expect(res.id).toEqual('id');
+                    done();
+                });
 
-            const reqLog = httpMock.expectOne('http://localhost:8080/api/auth/login');
-            expect(reqLog.request.method).toEqual('GET');
+                const reqLog = httpMock.expectOne('http://localhost:8080/api/auth/login');
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/authentication/services/methods/basic-authentication/basic-authentication.service.spec.ts:48]: yes, because subscription is called after request.flush

File: projects/netgrif-components-core/src/lib/authentication/services/methods/null-authentication/null-authentication.service.spec.ts [hunk 21]
```diff
@@ -1,6 +1,6 @@
-import {TestBed} from '@angular/core/testing';
+import {inject, TestBed} from '@angular/core/testing';
 import {ConfigurationService} from '../../../../configuration/configuration.service';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/authentication/services/methods/null-authentication/null-authentication.service.spec.ts:3]: done

[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/authentication/services/methods/null-authentication/null-authentication.service.spec.ts:3]: done

File: projects/netgrif-components-core/src/lib/authentication/sign-up/services/sign-up.service.spec.ts [hunk 22]
```diff
@@ -26,44 +26,58 @@ describe('SignUpService', () => {
...
-            })
+        it('should signup', (done) => {
+            inject([HttpTestingController],
+                    (httpMock: HttpTestingController) => {
+                        service.signup({token: 'string', name: 'string', surname: 'string', password: 'string'}).subscribe(res => {
+                            expect(res.success).toEqual('Done');
+                            done();
+                        });
+
+                        const reqLog = httpMock.expectOne('http://localhost:8080/api/auth/signup');
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/authentication/sign-up/services/sign-up.service.spec.ts:37]: yes, because subscription is called after request.flush

File: projects/netgrif-components-core/src/lib/header/abstract-header.component.spec.ts [hunk 23]
```diff
@@ -1,8 +1,7 @@
 import {waitForAsync, ComponentFixture, TestBed} from '@angular/core/testing';
 import {NoopAnimationsModule} from '@angular/platform-browser/animations';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/header/abstract-header.component.spec.ts:3]: done

File: projects/netgrif-components-core/src/lib/header/case-header/case-header.service.spec.ts [hunk 24]
```diff
@@ -7,7 +7,7 @@ import {HeaderColumn, HeaderColumnType} from '../models/header-column';
 import {SearchService} from '../../search/search-service/search.service';
 import {TestCaseBaseFilterProvider, TestCaseViewAllowedNetsFactory} from '../../utility/tests/test-factory-methods';
 import {CaseViewService} from '../../view/case-view/service/case-view-service';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/header/case-header/case-header.service.spec.ts:10]: done

File: projects/netgrif-components-core/src/lib/header/header-modes/search-mode/abstract-search-mode.component.spec.ts [hunk 25]
```diff
@@ -1,7 +1,7 @@
-import {waitForAsync, ComponentFixture, fakeAsync, TestBed, tick} from '@angular/core/testing';
+import {waitForAsync, ComponentFixture, fakeAsync, TestBed, tick, inject} from '@angular/core/testing';
 import {FlexLayoutModule, FlexModule} from '@angular/flex-layout';
 import {NoopAnimationsModule} from '@angular/platform-browser/animations';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/header/header-modes/search-mode/abstract-search-mode.component.spec.ts:4]: done

File: projects/netgrif-components-core/src/lib/header/header-modes/sort-mode/abstract-sort-mode.component.spec.ts [hunk 26]
```diff
@@ -1,8 +1,8 @@
-import {waitForAsync, ComponentFixture, TestBed} from '@angular/core/testing';
+import {waitForAsync, ComponentFixture, TestBed, inject} from '@angular/core/testing';
 import {FlexLayoutModule, FlexModule} from '@angular/flex-layout';
 import {MatSortModule} from '@angular/material/sort';
 import {NoopAnimationsModule} from '@angular/platform-browser/animations';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/header/header-modes/sort-mode/abstract-sort-mode.component.spec.ts:5]: done

File: projects/netgrif-components-core/src/lib/panel/task-panel/abstract-task-panel.component.spec.ts [hunk 27]
```diff
@@ -1,11 +1,11 @@
-import {ComponentFixture, TestBed, waitForAsync} from '@angular/core/testing';
+import {ComponentFixture, inject, TestBed, waitForAsync} from '@angular/core/testing';
 import {MatExpansionModule} from '@angular/material/expansion';
 import {CommonModule, CurrencyPipe} from '@angular/common';
 import {AfterViewInit, Component, Inject, Injector, NO_ERRORS_SCHEMA, ViewChild} from '@angular/core';
 import {NoopAnimationsModule} from '@angular/platform-browser/animations';
 import {Observable, of, Subject, throwError} from 'rxjs';
 import {map} from 'rxjs/operators';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/panel/task-panel/abstract-task-panel.component.spec.ts:8]: done

File: projects/netgrif-components-core/src/lib/process/public-process.service.spec.ts [hunk 28]
```diff
@@ -1,23 +1,27 @@
-import { TestBed } from '@angular/core/testing';
+import {TestBed} from '@angular/core/testing';
 
 import {PublicProcessService} from './public-process.service';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/process/public-process.service.spec.ts:4]: done

File: projects/netgrif-components-core/src/lib/resources/engine-endpoint/public/public-case-resource.service.spec.ts [hunk 29]
```diff
@@ -1,23 +1,27 @@
-import { TestBed } from '@angular/core/testing';
+import {inject, TestBed} from '@angular/core/testing';
 
-import { PublicCaseResourceService } from './public-case-resource.service';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {PublicCaseResourceService} from './public-case-resource.service';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/resources/engine-endpoint/public/public-case-resource.service.spec.ts:4]: done

File: projects/netgrif-components-core/src/lib/routing/view-service/view.service.spec.ts [hunk 30]
```diff
@@ -1,6 +1,9 @@
 import {ViewService} from './view.service';
 import {Component, Type} from '@angular/core';
 import {ViewEntry} from './model/view-entry';
+import {inject, TestBed} from '@angular/core/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/routing/view-service/view.service.spec.ts:5]: done

File: projects/netgrif-components-core/src/lib/search/category-factory/default-categories-factories.spec.ts [hunk 31]
```diff
@@ -1,11 +1,11 @@
 import {Inject, Injectable} from '@angular/core';
 import {NAE_SEARCH_CATEGORIES} from './search-categories-injection-token';
-import {TestBed} from '@angular/core/testing';
+import {inject, TestBed} from '@angular/core/testing';
 import {defaultCaseSearchCategoriesFactory, defaultTaskSearchCategoriesFactory} from './default-categories-factories';
 import {CategoryFactory} from './category-factory';
 import {TestConfigurationService} from '../../utility/tests/test-config';
 import {ConfigurationService} from '../../configuration/configuration.service';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/search/category-factory/default-categories-factories.spec.ts:8]: done

File: projects/netgrif-components-core/src/lib/search/fulltext-search-component/abstract-fulltext-search.component.spec.ts [hunk 32]
```diff
@@ -1,14 +1,15 @@
...
-import {ComponentFixture, fakeAsync, TestBed, tick, waitForAsync} from '@angular/core/testing';
+import {ComponentFixture, fakeAsync, inject, TestBed, tick, waitForAsync} from '@angular/core/testing';
 import {MaterialModule} from '../../material/material.module';
 import {TranslateLibModule} from '../../translate/translate-lib.module';
 import {NoopAnimationsModule} from '@angular/platform-browser/animations';
 import {NAE_BASE_FILTER} from '../models/base-filter-injection-token';
 import {TestCaseBaseFilterProvider} from '../../utility/tests/test-factory-methods';
 import {ConfigurationService} from '../../configuration/configuration.service';
 import {TestConfigurationService} from '../../utility/tests/test-config';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/search/fulltext-search-component/abstract-fulltext-search.component.spec.ts:12]: done

File: projects/netgrif-components-core/src/lib/search/search-predicate-component/abstract-search-predicate.component.spec.ts (hunk 1 above)
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/search/search-predicate-component/abstract-search-predicate.component.spec.ts:10]: done

File: projects/netgrif-components-core/src/lib/search/search-service/search.service.spec.ts (hunk 2 above)
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/search/search-service/search.service.spec.ts:8]: done

File: projects/netgrif-components-core/src/lib/side-menu/content-components/import-net/abstract-import-net.component.spec.ts (hunk 3 above)
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/side-menu/content-components/import-net/abstract-import-net.component.spec.ts:2]: done

File: projects/netgrif-components-core/src/lib/side-menu/side-menu-container/abstract-side-menu-container.component.spec.ts (hunk 4 above)
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/side-menu/side-menu-container/abstract-side-menu-container.component.spec.ts:4]: done

File: projects/netgrif-components-core/src/lib/task-content/task-content/abstract-task-content.component.spec.ts [hunk 33]
```diff
@@ -14,7 +14,7 @@ import {FieldConverterService} from '../services/field-converter.service';
 import {PaperViewService} from '../../navigation/quick-panel/components/paper-view.service';
 import {LoggerService} from '../../logger/services/logger.service';
 import {TaskEventService} from '../services/task-event.service';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/task-content/task-content/abstract-task-content.component.spec.ts:17]: done

File: projects/netgrif-components-core/src/lib/task/services/cancel-task.service.spec.ts [hunk 34]
```diff
@@ -1,8 +1,8 @@
-import {TestBed} from '@angular/core/testing';
+import {inject, TestBed} from '@angular/core/testing';
 import {CancelTaskService} from './cancel-task.service';
 import {MaterialModule} from '../../material/material.module';
 import {TranslateLibModule} from '../../translate/translate-lib.module';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/task/services/cancel-task.service.spec.ts:5]: done

File: projects/netgrif-components-core/src/lib/task/services/delegate-task.service.spec.ts (hunk 5 above)
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/task/services/delegate-task.service.spec.ts:6]: done

File: projects/netgrif-components-core/src/lib/task/services/finish-task.service.spec.ts [hunk 35]
```diff
@@ -1,8 +1,8 @@
-import {TestBed} from '@angular/core/testing';
+import {inject, TestBed} from '@angular/core/testing';
 import {FinishTaskService} from './finish-task.service';
 import {ConfigurationService} from '../../configuration/configuration.service';
 import {TestConfigurationService} from '../../utility/tests/test-config';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/task/services/finish-task.service.spec.ts:5]: done

File: projects/netgrif-components-core/src/lib/task/services/task-data.service.spec.ts (hunk 6 above)
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/task/services/task-data.service.spec.ts:5]: done

File: projects/netgrif-components-core/src/lib/view/case-view/abstract-case-view.spec.ts [hunk 36]
```diff
@@ -3,10 +3,10 @@ import {AbstractCaseView} from './abstract-case-view';
 import {AllowedNetsService} from '../../allowed-nets/services/allowed-nets.service';
 import {CaseViewService} from './service/case-view-service';
 import {Case} from '../../resources/interface/case';
-import {ComponentFixture, TestBed, waitForAsync} from '@angular/core/testing';
+import {ComponentFixture, inject, TestBed, waitForAsync} from '@angular/core/testing';
 import {MaterialModule} from '../../material/material.module';
 import {NoopAnimationsModule} from '@angular/platform-browser/animations';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
```
[the remark this answers is not in the document]
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/view/case-view/abstract-case-view.spec.ts:9]: done

**Review by minop — APPROVED** [review]

**Review by minop — APPROVED** [review]
